Automattic/jetpack
# Jetpack Search: Adapt modal styling from P2+ Project

#### Changes proposed in this Pull Request:
* Completely restyles the Jetpack Instant Search experience.
* Adds improved responsive handling of various viewports, including mobile and desktop.
* Removes theme and opacity customization.
* Moves form controls out of `SearchBox` and into `SearchForm`.

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
* Apply this change to your site with a Jetpack Search subscription. Alternatively, use Jurassic Ninja to spin up a test site.
* Enter a query into a search input. Ensure that the new modal appears.
* Try using different search result formats via the Customizer. You can also use `result_format=minimal`, `result_format=expanded`, and `result_format=product` query strings in your address bar to manually enable result formats.


#### Visual changes:

**Minimal Result Format**
> <img width="517" alt="Screen Shot 2021-02-03 at 5 16 16 PM" src="https://user-images.githubusercontent.com/4044428/106826916-18d01a00-6645-11eb-98b8-ae4f8cc46d4f.png">
> <img width="806" alt="Screen Shot 2021-02-03 at 5 15 31 PM" src="https://user-images.githubusercontent.com/4044428/106826927-1e2d6480-6645-11eb-9311-391cc994378a.png">
> <img width="774" alt="Screen Shot 2021-02-03 at 5 15 06 PM" src="https://user-images.githubusercontent.com/4044428/106826934-1ff72800-6645-11eb-9a77-65c9f4b8c477.png">

---
**Expanded Result Format**
> <img width="517" alt="Screen Shot 2021-02-03 at 5 16 12 PM" src="https://user-images.githubusercontent.com/4044428/106826918-1b327400-6645-11eb-91ed-17eb868ed46e.png">
> <img width="806" alt="Screen Shot 2021-02-03 at 5 15 35 PM" src="https://user-images.githubusercontent.com/4044428/106826924-1d94ce00-6645-11eb-832b-fe61f961fa19.png">
> <img width="774" alt="Screen Shot 2021-02-03 at 5 15 09 PM" src="https://user-images.githubusercontent.com/4044428/106826930-1f5e9180-6645-11eb-98d5-1ea0d36e95fc.png">

---
**Product Result Format**
> <img width="517" alt="Screen Shot 2021-02-03 at 5 16 08 PM" src="https://user-images.githubusercontent.com/4044428/106826920-1c63a100-6645-11eb-80ae-b34918bf6118.png">
> <img width="806" alt="Screen Shot 2021-02-03 at 5 15 41 PM" src="https://user-images.githubusercontent.com/4044428/106826922-1cfc3780-6645-11eb-9561-def0a9323468.png">
> <img width="774" alt="Screen Shot 2021-02-03 at 5 15 13 PM" src="https://user-images.githubusercontent.com/4044428/106826929-1ec5fb00-6645-11eb-9874-f196bfcc62fa.png">

---
**Expanded Result Format, Dark Theme**
> <img width="577" alt="Screen Shot 2021-02-09 at 3 00 02 PM" src="https://user-images.githubusercontent.com/4044428/107434167-9ed9de00-6ae7-11eb-8ac7-12449d04a794.png">
> <img width="900" alt="Screen Shot 2021-02-09 at 3 00 13 PM" src="https://user-images.githubusercontent.com/4044428/107434164-9da8b100-6ae7-11eb-9279-be2f41cd8ee4.png">
> <img width="900" alt="Screen Shot 2021-02-09 at 3 03 44 PM" src="https://user-images.githubusercontent.com/4044428/107434505-16a80880-6ae8-11eb-9df9-48c7b9a695f2.png">

#### Proposed changelog entry for your changes:
* Applies new, modal-based design to Jetpack Instant Search. This change is likely to break any custom CSS tweaks.
